app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -1603,17 +1425,21 @@ private void handleCloseButton() {
         return resultList;
     }
 
-    private static final class QueuedMedia {
-        Type type;
-        ProgressImageView preview;
-        Uri uri;
-        String id;
-        Call<Attachment> uploadRequest;
-        URLSpan uploadUrl;
-        ReadyStage readyStage;
-        byte[] content;
-        long mediaSize;
-        String description;
+    @Override
+    public void onEmojiSelected(@NotNull String shortcode) {
+        textEditor.getText().insert(textEditor.getSelectionStart(), ":"+shortcode+": ");
+    }
+
+    public static final class QueuedMedia {
```

Comment:
If it's not exclusive to this activity anymore we could move it outside

Reply:
good idea